src/averageAges.js

```diff
@@ -37,7 +44,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  const mothersNames = people.map(x => x.mother);
+
+  const mothers = women.filter(x => mothersNames.includes(x.name));
+
+  const womenAverageAge = women.reduce((sum, x) =>
+    sum + (x.died - x.born), 0) / women.length;
+
+  const mothersAverageAge = mothers.reduce((sum, x) =>
+    sum + (x.died - x.born), 0) / mothers.length;
+
+  return withChildren ? mothersAverageAge : womenAverageAge;
```

Comment:
Same here, move check for withChildren existence at the start and delete redundant reduce method